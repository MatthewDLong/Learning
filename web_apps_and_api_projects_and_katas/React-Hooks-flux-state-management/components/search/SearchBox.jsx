import React, { useContext } from "react";
import SearchBoxForm from "./SearchBoxForm";
import SearchResults from "./SearchResults";

import { SearchContext } from "../../state/context/search";

import { searchbox } from "./search.css";

const SearchBox = () => {
  const { searchResults } = useContext(SearchContext);

  return (
    <div className={searchbox}>
      <SearchBoxForm />
      {searchResults && <SearchResults />}
    </div>
  );
};

export default SearchBox;
