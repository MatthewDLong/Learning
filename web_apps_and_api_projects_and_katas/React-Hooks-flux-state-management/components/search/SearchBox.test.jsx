import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import SearchBox from "./SearchBox";

describe("SearchBox", () => {
  it("renders correctly", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<SearchBox />);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
