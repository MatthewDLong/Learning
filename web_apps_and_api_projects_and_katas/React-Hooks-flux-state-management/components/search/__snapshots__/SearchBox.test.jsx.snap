// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox renders correctly 1`] = `
<div
  className="searchbox"
>
  <SearchBoxForm />
</div>
`;
